libraries/chain/block_header_state_legacy.cpp

```diff
@@ -423,7 +423,7 @@ namespace eosio::chain {
       std::tie(is_satisfied, relevant_sig_count) = producer_authority::keys_satisfy_and_relevant(keys, valid_block_signing_authority);
 
       EOS_ASSERT(relevant_sig_count == keys.size(), wrong_signing_key,
-                 "block signed by unexpected key",
+                 "block signed by unexpected key: ${signing_keys}, expected: ${authority}. ${c} != ${s}",
```

Comment:
Looks like `${c}` and `${s}` are not provided.

Reply:
Fixed.